### ISSUE TITLE
[Bug]: scientific notation parse error

### ISSUE THREAD

**Issue by Dream95** [matrixorigin/matrixone#4957]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): v0.5.1
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> create table t1(c1 int);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t1 values(1.1e5);
Query OK, 1 row affected (0.02 sec)

mysql> select * from t1;
+------+
| c1   |
+------+
|    1 |
+------+
1 row in set (0.01 sec)

### Expected Behavior

mysql> create table t1(c1 int);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t1 values(1.1e5);
Query OK, 1 row affected (0.02 sec)

mysql> select * from t1;
+--------+
| c1   |
+--------+
|  110000 |
+--------+
1 row in set (0.01 sec)

**Comment by iamlinjunhong** [comment]
Not reproducible：
<img width="874" alt="截屏2022-09-19 下午3 11 47" src="https://user-images.githubusercontent.com/49111204/190966377-77d71fd9-cdea-4e65-8c3e-a7e4d58953fe.png">

**Comment by aressu1985** [comment]
fixed:fc685762b89cd931bdac2c09b93f6bab4ceeb6d5

mysql> create table t1(c1 int);
Query OK, 0 rows affected (0.01 sec)

mysql>  insert into t1 values(1.1e5);
Query OK, 1 row affected (0.01 sec)

mysql> select * from t1;
+--------+
| c1     |
+--------+
| 110000 |
+--------+
1 row in set (0.00 sec)
